src/usr/ipmi/ipmifruinv.C

```diff
@@ -925,12 +928,28 @@ errlHndl_t isdimmIpmiFruInv::addSpdDetails(std::vector<uint8_t> &io_data)
                             DEVICE_SPD_ADDRESS(SPD::MODULE_RANKS));
         if (l_errl) { break; }
 
+        uint8_t l_signalLoading = l_invalidValue;
+        l_errl = deviceRead(iv_target, &l_signalLoading, l_fieldSize,
+                            DEVICE_SPD_ADDRESS(SPD::SDRAM_DEVICE_TYPE_SIGNAL_LOADING));
+        if (l_errl) { break; }
+
+        uint8_t l_dieCount = 0;
+        if (l_signalLoading == JEDEC_SIGNAL_LOADING_3DS_TYPE)
+        {
+            l_dieCount = l_invalidValue;
+            l_errl = deviceRead(iv_target, &l_dieCount, l_fieldSize,
+                                DEVICE_SPD_ADDRESS(SPD::SDRAM_DIE_COUNT));
+            if (l_errl) { break; }
+        }
+
         // Calculate capacity of memory module
         uint16_t l_capacityGiB = 0;
         if (l_density <= JEDEC_DENSITY_MASK &&
             l_busWidthPri <= JEDEC_MEMORY_BUS_WIDTH_PRI_MASK &&
             l_dramWidth <= JEDEC_DRAM_WIDTH_MASK &&
-            l_ranks <= JEDEC_RANKS_MASK)
+            l_ranks <= JEDEC_RANKS_MASK &&
+            l_signalLoading <= JEDEC_SIGNAL_LOADING_MASK &&
```

Comment:
I'm seeing that checks are done for the 3DS values being at or under their maximum; however, no checks are done to ensure that the signal loading is at a proper value.  Per my version of the SPD spec, I'm seeing that signal loading only has valid values of 0b01 and 0b10.
Does this check need to be added into the code?

Reply:
Yes, I agree with you. I think that this check is not required.